tests/mock_crust/churn.rs

```diff
@@ -168,17 +168,22 @@ fn churn() {
         let auth_n1 = Authority::ManagedNode(nodes[index1].name());
         let auth_g0 = Authority::NaeManager(rng.gen());
         let auth_g1 = Authority::NaeManager(rng.gen());
+        let auth_c0 = Authority::ClientManager(rng.gen());
```

Comment:
A `ClientManager` is just a group like the other ones and isn't handled differently in routing, so this doesn't really add to the test.

What we'll need to test is `Authority::Client`: Sending from a client to a group (and possibly node) and vice-versa. That will require some modification to the `ExpectedGets`, and adding clients in addition to the nodes here in the first place.

We also need to make sure to either never churn out the clients' proxy nodes, or to somehow deal with reconnecting.